frontend/src/components/RoutePlanner.tsx

```diff
@@ -7,154 +7,78 @@ import {
-          prefixDataTestId="destination-input"
-        />
-      </IonItem>
-      <IonButton type="submit" size="default" expand="block">Search routes</IonButton>
-
-      {
-        originLocation !== null && destinationLocation !== null &&
-        <TripOptionsDisplay origin={originLocation} destination={destinationLocation} />
-      }
-    </IonList>
-  );
-
-};
-
-export function TripOptionsDisplay(props: { origin: Location, destination: Location }): JSX.Element {
-  const { origin, destination } = props;
-
-  const result = useJourneyQuery(origin, destination);
+  const currentIsFavoriteTrip = (): boolean => {
+    const existing = favoriteTrips.find(c =>
+      c.originId === originId
+      && c.destinationId === destinationId
+    );
+    return existing !== undefined;
+  };
 
-  const iJourneys: false | IJourney[] = result.type === "success" && result.journeyResults.map((journey): IJourney => {
-    const lastLeg = journey.legs[journey.legs.length - 1];
-    const firstLeg = journey.legs[0];
+  const canCurrentBeFavorited = (): boolean => originLocation !== null && destinationLocation !== null && !currentIsFavoriteTrip();
 
-    return {
-      startStation: firstLeg.origin.name,
-      startTime: firstLeg.origin.departure.estimated,
-      arrivalStation: lastLeg.destination.name,
-      arrivalTime: lastLeg.destination.arrival.estimated,
-      stops: journey.legs.map((leg): IJourneyStep => ({
-        arrivalTime: leg.destination.arrival.estimated,
-        startTime: leg.origin.departure.estimated,
-        stationName: leg.origin.name,
-        stopName: leg.destination.name,
-        track: "",
-        travelDurationInMinutes: leg.details.duration / 60
-      })),
-      travelDurationInMinutes: journey.legs.reduce((acc, leg) => acc + leg.details.duration, 0) / 60
-    };
-  });
+  const addToFavorites = (): void => {
+    if (originId === null || destinationId === null) return;
+    addFavoriteTrip({ originId, destinationId });
+  };
 
   return (
     <>
-      {result.type === "error" && <div>Error: {result.error.message}</div>}
-      {result.type === "pending" && <div>Searching...</div>}
-      {iJourneys &&
-        <JourneyListComponent journeys={iJourneys} />
-      }
+      <IonList inset={true}>
+        <IonItem lines="inset">
+          {/* Date-Time-Picker, allowing the user to select dates in the present aswell as the future */}
+          <IonLabel>Date and Time</IonLabel>
+          <IonDatetimeButton aria-label="Date and Time" datetime="datetime" />
+          <IonModal keepContentsMounted={true}>
+            <IonDatetime name="date_time" id="datetime" min={new Date().toISOString()} />
+          </IonModal>
+        </IonItem>
+        <IonItem>
+          <LocationSearchInput
+            inputLabel="Origin"
+            selectedLocation={originLocation}
+            onSelectedLocationChanged={(location): void => setOriginId(location.id)}
```

Comment:
I think this will cause an additional request, because:

1. The callback returns an stop object
2. setOriginId(...) will discard the object
3. There will be a rerender
4. useLocationByIdOrNull will then request the stop object for the given id

Reply:
(even thou there is a cache)

Reply:
You mean the location search wouldn't write to the cache? Yeah I know. But I thought maybe we should then adjust the backend type accordingly? Maybe it's time to get some own logic and data processing in the backend. All in all, out of scope for this PR?

Reply:
No. This is a regression introduced by this pr, so its inscope.

Reply:
Yes it is in scope, but there is nothing wrong with merging a functional pr and moving later implementations to a new branch. Otherwise we increasingly extend the lifetime of our branches and PRs, increasing the chances for merge conflicts. I would leave the issue open, but do this on a separate branch to get this going.